python/cudf/cudf/core/index.py

```diff
@@ -802,6 +803,16 @@ def second(self):
     def weekday(self):
         return self.get_dt_field("weekday")
 
+    @property
+    def dayofweek(self):
+        return self.weekday
+
+    def day_name(self):
+        return GenericIndex(wkday_idx_to_name(self.weekday))
```

Comment:
`self.weekday` will return you a column of integers, but the `wkday_idx_to_name` expects a scalar integer or a Python iterator to loop over. This will copy the data off the GPU element by element super inefficiently.